cpp/src/strings/repeat_strings.cu

```diff
@@ -0,0 +1,193 @@
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include <cudf/column/column_device_view.cuh>
+#include <cudf/column/column_factories.hpp>
+#include <cudf/detail/null_mask.hpp>
+#include <cudf/detail/nvtx/ranges.hpp>
+#include <cudf/strings/detail/utilities.cuh>
+#include <cudf/strings/detail/utilities.hpp>
+#include <cudf/strings/repeat_strings.hpp>
+#include <cudf/strings/strings_column_view.hpp>
+#include <cudf/utilities/error.hpp>
+
+#include <rmm/cuda_stream_view.hpp>
+
+#include <thrust/transform.h>
+
+namespace cudf {
+namespace strings {
+namespace detail {
+
+string_scalar repeat_strings(string_scalar const& input,
+                             size_type repeat_times,
+                             rmm::cuda_stream_view stream,
+                             rmm::mr::device_memory_resource* mr)
+{
+  if (!input.is_valid(stream)) { return string_scalar("", false, stream, mr); }
+  if (input.size() == 0 || repeat_times <= 0) { return string_scalar("", true, stream, mr); }
+  if (repeat_times == 1) { return string_scalar(input, stream, mr); }
+
+  CUDF_EXPECTS(input.size() <= std::numeric_limits<size_type>::max() / repeat_times,
+               "The output string has size that exceeds the maximum allowed size.");
+
+  auto const str_size = input.size();
+  auto const iter     = thrust::make_counting_iterator(0);
+  auto buff           = rmm::device_buffer(repeat_times * input.size(), stream, mr);
+
+  // Pull data from the input string into each byte of the output string.
+  thrust::transform(rmm::exec_policy(stream),
+                    iter,
+                    iter + repeat_times * str_size,
+                    static_cast<char*>(buff.data()),
+                    [in_ptr = input.data(), str_size] __device__(const auto idx) {
+                      return in_ptr[idx % str_size];
+                    });
+
+  return string_scalar(std::move(buff));
+}
+
+namespace {
+/**
+ * @brief Generate a strings column in which each row is an empty or null string.
+ *
+ * The output strings column has the same bitmask as the input column.
+ */
+auto generate_empty_output(strings_column_view const& input,
+                           size_type strings_count,
+                           rmm::cuda_stream_view stream,
+                           rmm::mr::device_memory_resource* mr)
+{
+  auto chars_column = create_chars_child_column(strings_count, 0, stream, mr);
+
+  static_assert(std::is_same_v<offset_type, int32_t> && std::is_same_v<size_type, int32_t>);
```

Comment:
```suggestion

```
This seems unnecessary. The code here should work if these types were to ever change.